src/devices/tpms_bmw_g3.c

```diff
@@ -64,40 +74,57 @@ static int tpms_bmwg3_decode(r_device *decoder, bitbuffer_t *bitbuffer)
 
     decoder_log_bitrow(decoder, 2, __func__, decoded.bb[0], decoded.bits_per_row[0], "DMC");
 
-    if (decoded.bits_per_row[0] < 88) {
+    // Based on the lenght define if Gen2 10 bytes, 81 bits or Gen3, 11 bytes, 89 bits.
```

Comment:
`// Based on the length decide if Gen2 (10 bytes, 81 bits) or Gen3 (11 bytes, 89 bits).`